Tests/UIKitTests/UITableViewExtensionsTests.swift

```diff
@@ -104,6 +104,22 @@ final class UITableViewExtensionsTests: XCTestCase {
         XCTAssertFalse(tableView.isValidIndexPath(invalidIndexPath))
     }
 
+    func testSafeScrollToIndexPath() {
+        let validIndexPathTop = IndexPath(row: 0, section: 0)
+        XCTAssertNoThrow(tableView.safeScrollToRow(at: validIndexPathTop, at: .top, animated: false))
```

Comment:
Your method doesn't `throw` anything, neither does `UITableView.scrollToRow(at:at:)`, so you wouldn't technically `catch` it with this `XCTAssertNoThrow`.

It will still work though because `XCTAssert...` will report certain exceptions and continue running, but this isn't a valid test. E.g. You couldn't reverse it by calling
```XCTAssertThrowsError(tableView.scrollToRow(at: invalidIndexPath, at: .top, animated: false))```

You can either just calls these methods without `XCTest...` and test their side-effects (like you're already doing) or you can tell change the function to `throw` an error if the `indexPath` is invalid.

Reply:
Thanks for the input.
I'd rather go with the first choice, since I wan't this function to fail silently if `IndexPath` is invalid.